### PR TITLE
[PT-166471712] Add state diagrams for state channel fsm

### PR DESCRIPTION
**Pull request by tolbrino**
This is a draft for updating the FSM state diagram documentation.

The included source code changes are merely cosmetic to make the structure easier to understand. No logic has been changed.

### CODE REVIEW

File: docs/state-channels/fsm_initiator.puml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+@startuml
```
Comment (lucafavatella) [inline docs/state-channels/fsm_initiator.puml]: Would it be possible to automate as Makefile the generation of this plantuml file, and hook that to CI?

File: docs/state-channels/fsm_responder.puml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+@startuml
```
Comment (lucafavatella) [inline docs/state-channels/fsm_responder.puml]: Would it be possible to automate as Makefile the generation of this plantuml file, and hook that to CI?

File: Makefile [hunk 3]
```diff
@@ -512,6 +520,14 @@ $(DEB_PKG_CHANGELOG_FILE):
 prod-deb-package: $(DEB_PKG_CHANGELOG_FILE)
 	debuild --preserve-envvar DEB_SKIP_DH_AUTO_CLEAN -b -uc -us
 
+$(PLANTUML_JAR):
+	curl -fsS --create-dirs -o $@ https://netcologne.dl.sourceforge.net/project/plantuml/${PLANTUML_V}/plantuml.${PLANTUML_V}.jar
+
+build-uml: ${PLANTUML_JAR} ${uml-svg-files}
```
Comment (lucafavatella) [inline Makefile]: Would it be possible to hook this (puml -> svg) to CI so to prevent it not to be working in the future and go unnoticed?

Comment (tolbrino) [reply Makefile]: Sure, we could do that.

File: .circleci/config.yml [hunk 4]
```diff
@@ -946,6 +945,20 @@ jobs:
...
 
+  build_uml_diagrams:
+    executor: builder_container_stable
+    <<: *container_config
+    steps:
+      - checkout
+      - run:
+          name: Install PlantUML dependencies
+          command: |
+            sudo apt-get -qq -y update && sudo apt-get -qq -y install graphviz librsvg2-bin
```
Comment (lucafavatella) [inline .circleci/config.yml]: @tolbrino @dincho Shall this rather go in https://github.com/aeternity/docker-builder/blob/d15fa8e500d4ec4ee808aac6c3ea89a5b42c8f22/Dockerfile ?

Comment (lucafavatella) [reply .circleci/config.yml]: (Eventually...)

Comment (tolbrino) [reply .circleci/config.yml]: As per @dincho 's suggestion I used another pre-built image, which works fine.

File: .circleci/config.yml [hunk 5]
```diff
@@ -1443,6 +1457,18 @@ workflows:
             tags:
               only: *stable_tag_regex
 
+      - build_uml_diagrams:
+          requires:
+            - build
+          filters:
+            branches:
```
Comment (lucafavatella) [inline .circleci/config.yml]: It feels this should be only master.

**Pull request merged**